### ISSUE TITLE
Some Datagrid buttons are missing button names

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3371]
Some of the buttons in the datagrid are just announced as 'button' with a screen reader, and we need to add meaningful names to them.

* [ ] The close button in the clr-dg-column-toggle component lacks a mechanism for providing an accessible name
* [ ] The toggle button inside the clr-dg-column-toggle component lacks a mechanism for providing an accessible name
* [ ] There is no way for the developer using Clarity to provide an accessible name for the datagrid-action-toggle button
